pandas/tests/computation/test_eval.py

```diff
@@ -894,14 +895,17 @@ def testit(r_idx_type, c_idx_type, index_name):
                 expected = df + s
             tm.assert_frame_equal(res, expected)
 
-        with warnings.catch_warnings(record=True):
-            warnings.simplefilter("always", RuntimeWarning)
-
-            testit(r_idx_type, c_idx_type, index_name)
-
     @pytest.mark.parametrize("index_name", ["index", "columns"])
-    def test_basic_series_frame_alignment(self, engine, parser, index_name):
-        def testit(r_idx_type, c_idx_type, index_name):
+    @pytest.mark.parametrize(
+        "r_idx_type, c_idx_type",
+        list(product(["i", "u", "s"], ["i", "u", "s"])) + [("dt", "dt")],
+    )
+    def test_basic_series_frame_alignment(
+        self, engine, parser, index_name, r_idx_type, c_idx_type
+    ):
+        with warnings.catch_warnings(record=True):
+            # avoid warning about comparing strings and ints
+            warnings.simplefilter("ignore", RuntimeWarning)
```

Comment:
any particular reason you use `warnings` rather than `pytest.mark.filterwarnings`? (just interested)

Reply:
This was just a direct move from the old test. Refactored to use `pytest.mark.filterwarnings` (which appears what we now use)